packages/components/organisms/f-form/src/components/_tests/Form.test.js

```diff
@@ -76,17 +96,205 @@ describe('Form', () => {
+            let wrapper;
+
+            beforeEach(() => {
+                isFormValidSpy = jest.spyOn(FForm.methods, 'isFormValid');
+                scrollToFirstInlineErrorSpy = jest.spyOn(FForm.methods, 'scrollToFirstInlineError');
+
+                // Arrange
+                wrapper = shallowMount(FForm, { propsData });
+            });
 
+            afterEach(() => {
+                jest.clearAllMocks();
+            });
+
+            it('should emit with `FormSubmitting` event', () => {
                 // Act
                 wrapper.vm.onFormSubmit();
 
                 // Assert
                 expect(wrapper.emitted('form-submitting').length).toBe(1);
             });
+
+            describe('when `isFormValid` is true', () => {
+                beforeEach(() => {
+                    // Arrange
+                    isFormValidSpy.mockReturnValue(true);
+
+                    wrapper = shallowMount(FForm, { propsData });
+                });
+
+                afterEach(() => {
+                    jest.clearAllMocks();
+                });
+
+                it('should emit `FormValid` event with valid fields', () => {
+                    // Act
+                    wrapper.vm.onFormSubmit();
+
+                    // Assert
+                    expect(wrapper.emitted('form-valid').length).toBe(1);
+                    expect(wrapper.emitted('form-valid')[0][0]).toMatchSnapshot();
+                });
+            });
+
+            describe('when `isFormValid` is false', () => {
+                beforeEach(() => {
+                    // Arrange
+                    isFormValidSpy.mockReturnValue(false);
+
+                    wrapper = shallowMount(FForm, { propsData });
+                });
+
+                afterEach(() => {
+                    jest.clearAllMocks();
+                });
+
+                it('should call `scrollToFirstInlineError` and emit `FormInvalid` event', () => {
+                    // Act
+                    wrapper.vm.onFormSubmit();
+
+                    // Assert
+                    expect(scrollToFirstInlineErrorSpy).toHaveBeenCalled();
+                    expect(wrapper.emitted('form-invalid').length).toBe(1);
+                    expect(wrapper.emitted('form-invalid')[0][0]).toMatchSnapshot();
+                });
+            });
+        });
+
+        describe('isValidPhoneNumber ::', () => {
+            afterEach(() => {
+                jest.clearAllMocks();
+            });
+
+            it('should call `isValidPhoneNumber` from `f-services`', () => {
+                // Arrange
+                const isValidPhoneNumberSpy = jest.spyOn(validations, 'isValidPhoneNumber');
+
+                const wrapper = shallowMount(FForm, { propsData });
+
+                // Act
+                wrapper.vm.isValidPhoneNumber();
+
+                // Assert
+                expect(isValidPhoneNumberSpy).toHaveBeenCalledWith(formData.formFields.mobileNumber.value, 'en-GB');
```

Comment:
For example, this is a good use of a spy.
It's an external method so you can't just stub it with `jest.fn` (unless you use a module mock which can get a bit hairy), and you're also using an expectation like `toHaveBeenCalledWith`.